lib/pure/strutils.nim

```diff
@@ -94,6 +95,23 @@ const
   Letters* = {'A'..'Z', 'a'..'z'}
     ## The set of letters.
 
+  UpperCaseLetters* = {'A'..'Z'}
+    ## The set of uppercase letters.
+
+  LowerCaseLetters* = {'a'..'z'}
+    ## The set of lowercase letters.
+
+  Punctuations* = {'!'..'/', ':'..'@', '['..'`', '{'..'~'}
+    ## The set of all special symbols.
+
+  PrintablesNoWhiteSpace* = {'!'..'~'}
+    ## The set of all printable characters (letters, digits,
+    ## punctuations). No Whitespace.
+
+  Printables* = PrintablesNoWhiteSpace + Whitespace
+    ## The set of all printable characters (letters, digits,
+    ## punctuations, Whitespace).
+
```

Comment:
- `UpperCase`, `LowerCase`, and `Whitespace` should be `Uppercase`, `Lowercase`, and `Whitespace`, respectively, as they are all single words.
- Use `PrintableChars`, `PunctuationChars`, etc. rather than just `Printables`.
- Comments should be added to mention that these are sets of `ASCII` characters (yes, this should already be known, but a little clarification shouldn't hurt).
- I feel that `PrintablesNoWhitespace` is unnecessary - one can use something like `Printables - Whitespace` in the rare case this is needed. It can always be added later on if it turns out there is a pronounced need for it.

```suggestion
  UppercaseLetters* = {'A'..'Z'}
    ## The set of uppercase ASCII letters.

  LowercaseLetters* = {'a'..'z'}
    ## The set of lowercase ASCII letters.

  PunctuationChars* = {'!'..'/', ':'..'@', '['..'`', '{'..'~'}
    ## The set of all ASCII punctuation characters.

  PrintableChars* = Letters + Digits + PunctuationChars + Whitespace
    ## The set of all printable characters (letters, digits, whitespace, and punctuation characters).

```